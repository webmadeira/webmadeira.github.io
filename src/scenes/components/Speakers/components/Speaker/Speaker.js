import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { H3, Body2 } from '../../../../../components/typography'

const Root = styled.div`
  padding: 5px;
`

const Photo = styled.img`
  border-radius: 50%;
  border: 2px solid ${props => props.theme.color()};
  width: 150px;
  
  ${({ theme }) => theme.mixins.mediaQuery.tablet`
    width: 100px;
  `};

  ${({ theme }) => theme.mixins.mediaQuery.phone`
    width: 70px;
  `};
`

const Name = styled.div`
    padding: 10px 0;
`

const Job = styled.div`
  text-align: center;
  color: ${props => props.theme.color('neutral', 'base')};
`

const Link = styled.a`
  color: #535353;
  text-decoration: none;
`

const photoOnFocusCb = (e, speaker) => {
  e.currentTarget.src = speaker.photo
}

const photoOnBlurCb = (e, speaker) => {
  e.currentTarget.src = speaker.bwPhoto
}

const Speaker = ({ speaker }) => (
  <Root key={speaker.id}>
    {/* Prefetch the image to avoid downloading it only when "hover" occurs */}
    <link rel="prefetch" href={speaker.photo} />
    <Photo
      src={speaker.bwPhoto}
      alt="Speaker"
      onMouseOver={e => photoOnFocusCb(e, speaker)}
      onFocus={e => photoOnFocusCb(e, speaker)}
      onBlur={e => photoOnBlurCb(e, speaker)}
      onMouseOut={e => photoOnBlurCb(e, speaker)}
    />
    <Name>
      <H3>
        <Link href={speaker.personalWebsite}>{speaker.name}</Link>
      </H3>
    </Name>
    <Job>
      <Body2>
        {speaker.jobTitle},
      </Body2>
      <br />
      <Body2>
          at <Link href={speaker.jobWebsite}>{speaker.company}</Link>
      </Body2>
    </Job>
  </Root>
)

Speaker.propTypes = {
  speaker: PropTypes.shape({
    id: PropTypes.number.isRequired,
    bwPhoto: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    personalWebsite: PropTypes.string.isRequired,
    jobTitle: PropTypes.string.isRequired,
    company: PropTypes.string.isRequired,
  }).isRequired,
}

export default Speaker
