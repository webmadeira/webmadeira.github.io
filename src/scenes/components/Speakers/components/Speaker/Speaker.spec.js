import React from 'react'
import { shallow } from 'enzyme'
import Speaker from './Speaker'

describe('Speaker component', () => {
  it('must render properly', () => {
    const speaker = {
      id: 0,
      name: 'André Gonçalves',
      company: 'Freelancer',
      jobTitle: 'Web Dev Consultant',
      personalWebsite: 'https://github.com/andregoncalves',
      bwPhoto: 'https://avatars3.githubusercontent.com/u/1074?s=460&v=4',
      photo: 'https://avatars3.githubusercontent.com/u/1074?s=460&v=4',
    }

    const component = shallow(<Speaker speaker={speaker} />)

    expect(component).toMatchSnapshot()
  })
})
