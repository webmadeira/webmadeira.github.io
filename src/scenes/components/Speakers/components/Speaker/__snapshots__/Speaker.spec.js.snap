// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Speaker component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Speaker
    speaker={
      Object {
        "bwPhoto": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
        "company": "Freelancer",
        "id": 0,
        "jobTitle": "Web Dev Consultant",
        "name": "André Gonçalves",
        "personalWebsite": "https://github.com/andregoncalves",
        "photo": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "0",
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <link
          href="https://avatars3.githubusercontent.com/u/1074?s=460&v=4"
          rel="prefetch"
        />,
        <styled.img
          alt="Speaker"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          src="https://avatars3.githubusercontent.com/u/1074?s=460&v=4"
        />,
        <styled.div>
          <styled.h3>
            <styled.a
              href="https://github.com/andregoncalves"
            >
              André Gonçalves
            </styled.a>
          </styled.h3>
        </styled.div>,
        <styled.div>
          <styled.span>
            Web Dev Consultant
            ,
          </styled.span>
          <br />
          <styled.span>
            at 
            <styled.a
              href={undefined}
            >
              Freelancer
            </styled.a>
          </styled.span>
        </styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "href": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
          "rel": "prefetch",
        },
        "ref": null,
        "rendered": null,
        "type": "link",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alt": "Speaker",
          "onBlur": [Function],
          "onFocus": [Function],
          "onMouseOut": [Function],
          "onMouseOver": [Function],
          "src": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.h3>
            <styled.a
              href="https://github.com/andregoncalves"
            >
              André Gonçalves
            </styled.a>
          </styled.h3>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.a
              href="https://github.com/andregoncalves"
            >
              André Gonçalves
            </styled.a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "André Gonçalves",
              "href": "https://github.com/andregoncalves",
            },
            "ref": null,
            "rendered": "André Gonçalves",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.span>
              Web Dev Consultant
              ,
            </styled.span>,
            <br />,
            <styled.span>
              at 
              <styled.a
                href={undefined}
              >
                Freelancer
              </styled.a>
            </styled.span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Web Dev Consultant",
                ",",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Web Dev Consultant",
              ",",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "at ",
                <styled.a
                  href={undefined}
                >
                  Freelancer
                </styled.a>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "at ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Freelancer",
                  "href": undefined,
                },
                "ref": null,
                "rendered": "Freelancer",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "0",
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <link
            href="https://avatars3.githubusercontent.com/u/1074?s=460&v=4"
            rel="prefetch"
          />,
          <styled.img
            alt="Speaker"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            src="https://avatars3.githubusercontent.com/u/1074?s=460&v=4"
          />,
          <styled.div>
            <styled.h3>
              <styled.a
                href="https://github.com/andregoncalves"
              >
                André Gonçalves
              </styled.a>
            </styled.h3>
          </styled.div>,
          <styled.div>
            <styled.span>
              Web Dev Consultant
              ,
            </styled.span>
            <br />
            <styled.span>
              at 
              <styled.a
                href={undefined}
              >
                Freelancer
              </styled.a>
            </styled.span>
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "href": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
            "rel": "prefetch",
          },
          "ref": null,
          "rendered": null,
          "type": "link",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "Speaker",
            "onBlur": [Function],
            "onFocus": [Function],
            "onMouseOut": [Function],
            "onMouseOver": [Function],
            "src": "https://avatars3.githubusercontent.com/u/1074?s=460&v=4",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.h3>
              <styled.a
                href="https://github.com/andregoncalves"
              >
                André Gonçalves
              </styled.a>
            </styled.h3>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <styled.a
                href="https://github.com/andregoncalves"
              >
                André Gonçalves
              </styled.a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "André Gonçalves",
                "href": "https://github.com/andregoncalves",
              },
              "ref": null,
              "rendered": "André Gonçalves",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.span>
                Web Dev Consultant
                ,
              </styled.span>,
              <br />,
              <styled.span>
                at 
                <styled.a
                  href={undefined}
                >
                  Freelancer
                </styled.a>
              </styled.span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "Web Dev Consultant",
                  ",",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Web Dev Consultant",
                ",",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "at ",
                  <styled.a
                    href={undefined}
                  >
                    Freelancer
                  </styled.a>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "at ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Freelancer",
                    "href": undefined,
                  },
                  "ref": null,
                  "rendered": "Freelancer",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
